frame/support/procedural/src/storage/storage_struct.rs

```diff
@@ -207,9 +207,107 @@ pub fn decl_and_impl(scrate: &TokenStream, def: &DeclStorageDefExt) -> TokenStre
 			}
 		};
 
+		let max_values = if let Some(max_values) = &line.max_values {
+			quote::quote!({
+				let max_values: u32 = (|| #max_values)();
+				Some(max_values)
```

Comment:
I don't understand this idiom. Why not just

```rust
Some(#max_values)
```

?

Reply:
I was thinking if someone use a weird expression with some `return` inside, it is better to put it inside a function

Reply:
Huh, ok. Better safe than sorry!